helios-client/src/main/java/com/spotify/helios/client/AuthenticatingHttpConnector.java

```diff
@@ -105,8 +111,11 @@ public HttpURLConnection connect(final URI uri, final String method, final byte[
     try {
       log.debug("connecting to {}", ipUri);
 
-      if (clientCertificatePath.isPresent()) {
-        // prioritize using the certificate file if set
+      if (accessTokenOpt.isPresent()) {
+        // prioritize using bearer token
+        return connectWithBearerToken(ipUri, method, entity, headers);
+      } else if (clientCertificatePath.isPresent()) {
```

Comment:
I think one benefit of sending both would be that we could merge and release this ahead of any ability in the backend/server-side to handle these tokens, to decouple the two.